### ISSUE TITLE
Ask and answer gardening questions

### ISSUE THREAD

**Issue by sophiakc** [openfarmcc/OpenFarm#863]
The typical interface would be a Stackoverflow-like place where gardeners can ask question, answers, rank/ star the most relevant ones to appear first.

According to @simonv3, this would preferably be built natively, rather than implementing (and maintaining) an external solution.

As a 2nd step, this content will be computed in the result from the search on Openfarm.

**Comment by sophiakc** [comment]
What about using Github Wiki for this purpose? Cc @simonv3

**Comment by simonv3** [comment]
The issue I have with using a different service is that it's not integratabtle in a significant way. Unless we start creating some format of "smart" links we don't know what's on the linked page, and we also can't guarantee that the things there stay the same. 

Also, the GitHub Wiki doesn't really suit this kind of purpose? There's too much "GitHub" knowledge involved to just ask a question? How do you keep track of questions that were asked? How do people find the specific question they're hoping has been asked already?

**Comment by sophiakc** [comment]
I have done this class and they have all their documentation on a wiki: http://guides.codepath.com/ios/Designer-Guides
It does the job and doesn't require people to necessary knows how to use Github.

There is a need of something simple to implement. If not Github wiki, what would you suggest?

**Comment by simonv3** [comment]
I wonder if @andru and @roryaronson can pitch in on this.

**Comment by andru** [comment]
Funnily enough I'm currently contracting on a gardening mobile app which works around a Q&A model.

There are plenty of OS options of varying quality, but [QSQA](http://www.dzonesoftware.com/products/open-source-question-answer-software) is a popular one.

Not a small undertaking any which way it's handled, really, if we want to allow users to answer each other's questions.

**Comment by andru** [comment]
For a closer tech stack to ours, there's also [OpenOverflow](https://github.com/nsanta/openoverflow) but I've never played with it.

**Comment by roryaronson** [comment]
Sorry for the late chime in. I think Q/A functionality would be cool eventually, though I don't think using a third-party platform would be good for the future compared to something built-in to OpenFarm natively. I think it also makes sense for any Q/A functionality to initially revolve around a piece of content such as a Guide, rather than free standing. So the person with the question would be asking it to the Guide author and the other readers of that Guide, and either the author or other readers could chime in. If it the questions were free standing it would seem more like a forum, of which there are many gardening forums available already.

So I think that this functionality could be achieved in the form of a commenting system on the Guides. Then the comments are linked to user profiles, and we can be sending emails out to the guide author when they get a question/comment.

**Comment by sophiakc** [comment]
Hey @roryaronson 
I got your arguments.
It sounds to me that the perfect solution is wanted rather than a quick and dirty one, that would provide knowledge about how users are behaving and what they are looking for. This kind of data is really missing.
What about using Slack for that? A specific one isolated from the one of the developers.

**Comment by roryaronson** [comment]
Correct. I think any type of forum/chatroom stuff doesn't need to be built because there are already venues for that online and off. I see the social elements of OpenFarm revolving around content, at least initially.

**Comment by CloCkWeRX** [comment]
With growstuff, I think its fair to say we've got more developers talking about gardening code on github... than gardeners themselves via our forums - there doesn't seem to be a huge overlap between 'people who want to garden' and 'people who want to talk about gardening and have questions'.


It might be worth looking at some active community gardens and how they manage things - newsletters, facebook, etc seem to work well 'enough' for a lot of folks; even if some of the platforms are non-free/non-open; and don't really promote Q&A well.

As an example, a nearby community garden:
![image](https://cloud.githubusercontent.com/assets/365751/22868800/bd3d9a06-f1e9-11e6-8db5-35d274ae9973.png)

 * Highly visual posts
 * Comments don't seem to be discussion oriented really
 * 2/797 followers commenting is a fairly low interaction rate, but probably not uncommon!
 * Most liked recent post was about the Christmas party, not so much about gardening

I would suspect because most of the people involved see each other on occasion at the same place, they don't need communication tools as much as a distributed community does. Going to guess that there would be few < 30 year olds in that group.

Another example:
https://gncg.wordpress.com/ - run by a friend of mine. It's a broadcast style of interaction; most communication is face to face in that group.
There's a bit more highlighting hard work done by people (ie see bits about new irrigation systems); but not very much on *how you actually grow particular things well*, which seems odd.

**Comment by sophiakc** [comment]
@roryaronson @CloCkWeRX What about opening a Slack for gardeners? So we could interact directly with gardeners and the way they create guides and all.

Thx for this usability research @CloCkWeRX I agree with your observation about face-to-face info sharing

**Comment by roryaronson** [comment]
I'm not convinced that an open chatroom is needed or would be effective for people. I personally think any type of free-form conversation needs to be much more focused to be useful. Eg: I would join a group that is for gardeners in my town, or with a small group of friends/acquaintances, or for "stone fruit growers" because that's a very specific thing I am interested in. I would not join a general chatroom because I think it would be very difficult/intimidating to connect and it would be unlikely that I received the help I need or could help others. There would be much more noise than useful conversation. We get a taste of this in our developer slack room already and in my opinion its largely unproductive conversation.

That's why I think any type of commenting/forum stuff needs to revolve around a piece of content such as a specific Growing Guide or Crop. Then people can have focused conversations with the right people.

**Comment by sophiakc** [comment]
@roryaronson 
I didn't say the slack should have only a general channel. People are free to create specific channels, and so we could see what they create. The use case you describe are very useful. But we can't imagine them until there is something out there.
Which tool, if not Slack, should we use to quickly get a sense of those use cases? That we could be live quickly? 

I think the comments on guide are one possible answer, but adds a lot of contraints, the 1st being the implementation time. 

--

If the Slack for developers is unproductive why do we use it? I agree on that

**Comment by roryaronson** [comment]
I think Slack for developers is useful when it is used by developers for development. When someone comes in to ask for general or specific gardening advice, in my opinion it is the wrong place and somewhat the wrong audience.

If we want to get a sense of what people are asking and helping each other out with, we can look at any of the existing gardening forums out there: https://www.google.com/search?q=gardening%20forum

Again I don't see much value in making another free-for-all chatroom or forum. I see a lot of value though in facilitating discussions around specific content with the right people: Commenting on Guides. And I think the success and usefulness of the commenting would actually be because of the constraints of this system. There is a topic (the Guide content) which keeps comments and replies focused. And because the dialogue is happening between people using that specific guide and the author of that specific guide, the matchmaking is strong between those who need help and those who can provide help.

**Comment by sophiakc** [comment]
@simonv3 said: 
We can set up a slack, but I’m personally not going to invest time into being active on it. If we can find another volunteer to do so, that’d be great!
I just don’t want to split audience / effort over this

=> I don't see the Slack for gardeners as a channel that we need to facilitate. It's more to let it lives by itself and observe what people ask and what they do with it. Other forums won't get a sense of what people want to do with Openfarm.

@roryaronson said:
To further my point on GitHub about a Slack for gardeners, I think the most recent comment in #general by blueies drives my point home. Nobody here is really equipped to help this person out and the vast majority of non-development comments are similar to that one. So the question asker is wasting their time askign the wrong people in the wrong place. And the people who are here are unable to help because the questions are too unfocused and often times out of our domain of expertise. Having interaction take place around a very specific topic (a Guide) with very specific people (the Guide author and other people looking at the Guide) would be much more effective and useful for both the question asker and the helpers.

=> I didn't say that gardeners should use the same slack than the developers, I agree that they should not be mixed, this is why I suggested a dedicated slack for them. 
I don't see this Slack thing as a final solution, but rather as an observatory tool ready to be used for the specific use case of Openfarm.
Right now, we don't have comment section on guides. This is why I asked, instead of a Slack, what OTHER TOOLS do you think of to observe the specific use cases and needs of gardeners on OpenFarm, and that is also fast to implement?

**Comment by sophiakc** [comment]
@roryaronson ? :)

**Comment by roryaronson** [comment]
I can't think of any other tools that I think would be worthwhile implementing to observe specific use cases or to make up for OpenFarm's currently lacking features. I think over the years we've received a lot of feedback and feature requests, and learned a lot about use cases and needs, etc. Now, our time is best spent actually implementing the features we're pretty certain will be useful rather than hacking around them trying to gain more understanding.

**Comment by sophiakc** [comment]
@roryaronson  
I get your point about feedback, features' requests, and needs. But this is one thing to collect them upfront and another to see them in action. And I don't have any ways to measure who is coming/ leaving/ willing to ask questions. 

Can't we give a try to a dedicated Slack for gardeners during 2 months? And based on its outcome, we can decide whether to continue or to close it. I am down for initiating it and actively interacting with people there if needed.
